src/Aspire.Hosting.Azure/AzureBicepResource.cs

```diff
@@ -29,6 +38,19 @@ public AzureBicepResource(string name, string? templateFile = null, string? temp
         TemplateResourceName = templateResourceName;
 
         Annotations.Add(new ManifestPublishingCallbackAnnotation(WriteToManifest));
+
+        // Add pipeline step annotation to provision this bicep resource
+        Annotations.Add(new PipelineStepAnnotation((factoryContext) =>
+        {
+            var provisionStep = new PipelineStep
+            {
+                Name = $"provision-{name}",
+                Action = async ctx => await ProvisionAzureBicepResourceAsync(ctx, this).ConfigureAwait(false)
+            };
+            provisionStep.RequiredBy(WellKnownPipelineTags.ProvisionInfrastructure);
```

Comment:
Create a new `WellKnownPipelineSteps` static class and define the `ProvisionInfrastructure` step name in there.